### DISCUSSION TITLE
Resilience of Poloniex web socket stream

### DISCUSSION THREAD

**Discussion by gayansm** [DigitalRuby/ExchangeSharp#755] [Q&A]
Hi,
I'm using ExchangeSharp in a project to get pricing from Poloniex. We primarily use the web socket ticker but fallback to the API (GetTickersAsync) if we detect that no prices were received for 5s. This setup has been working OK most of the time, but we see that there are quite a few instances where we don't get updates from the web socket for 5 seconds.

This behavior is only happening once we deploy the application to the cloud, which may be an indication that it is **related to CPU usage**. But we are using another package to get prices from Binance and it doesn't show any gaps in pricing under heavier loads.
I just want to get **your ideas on possible improvements** I can make to the web socket price receiver to make it faster and more resilient. I haven't changed any of the **default parameters** when setting up the connections, so may be those could be changed?

**Comment by jjxtra** [comment]
Hard to say unless you are profiling exchangesharp specifically and can point to a piece of code that needs to be optimized.
